### DISCUSSION TITLE
Discuss about spotify-api.js

### DISCUSSION THREAD

**Discussion by abh80** [spotify-api/spotify-api.js#14] [General]
Start your discussions can contain future updates !

**Comment by olivoreo** [comment]
Hello! I wanted to ask, what does the "tracks.search" action return? It means what the response text will be. I just wanted to make my app give me a link to the song found through the search. Can you tell me how to do this?

**Reply by abh80** [reply]
Hi,
The `track.search` action returns track search results which is an array

**Reply by olivoreo** [reply]
> Hi,
> The `track.search` action returns track search results which is an array

And how do I get links to tracks from this result?

**Reply by abh80** [reply]
You can get the href of track by extracting the href property from the track structure - Example 
```js
tracks[0].href
```

**Reply by olivoreo** [reply]
Of you a huge thank you! You've been very helpful!!!

**Reply by abh80** [reply]
No problem! Just feel free to reach out for help, I would be glad to assist you!
Also we are updating the documentation soon

**Reply by scientific-dev** [reply]
No discussions yet

**Comment by adb-sh** [comment]
Hey :)
i have some problems with the refresh tokens. Not sure if it's a bug or just me doing something wrong.
After about one hour form when I started the client I get a "access token expired" error message from the spotify api.
I'm creating the spotify client like this: (pseudo code)
```javascript
const params = new URLSearchParams();
params.append('grant_type', 'authorization_code');
params.append('code', code);
params.append('redirect_uri', store.redirectURL);

const config = {
  headers: {
    'Authorization': 'Basic ' + (new Buffer(store.clientID + ':' + store.clientSecret).toString('base64')),
    'Content-Type': 'application/x-www-form-urlencoded',
  },
};

const tokens = (await axios.post('https://accounts.spotify.com/api/token', params, config))?.data;

await Client.create({
  refreshToken: true,
  retryOnRateLimit: true,
  token: {
    clientID: store.clientID,
    clientSecret: store.clientSecret,
    redirectURL: store.redirectURL,
    refreshToken: tokens.refresh_token,
  },
  async onRefresh() {
    await UserStore.findOneAndUpdate(
      { 'spotify.userId': client.user.id },
      { spotify: { refreshToken: client.refreshMeta.refreshToken } },
    );
  },
});
```
I also tried to pass `code` directly without manually requesting a refresh token from the spotify api before.
The client acctually stores the the correct refreshToken in client.refreshMeta.refreshToken, but it doesn't automatically requests a new token, when needed apparently.

**Reply by adb-sh** [reply]
also, another weird thing i noticed is that i can't catch exceptions (f/e/ if Client.create fails to login) in a try/catch scope. Instead I receive a top level "unhandled Promise rejection" error message.
